rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,558 @@
+    """ Handles ligand labeling (conf label and atom label) in update3D function """
+    
+    uid = molViewState.uid
+    wgData = globals()['wgData'][uid]
+    sMolNames = molViewState.selectedMolNames
+    sConfIds = molViewState.selectedConfIds
+    
+    if not len(sMolNames)==1 and len(sConfIds)==1:
+        return None
+    else:
+        if 'confLabel' in wgData:
+            if wgData['confLabel'].value:
+                confLabel = list(sMolNames)[0] + ':' + str(wgData['confId'].value)
+                view.addLabel(confLabel, {'backgroundColor':'gray', 'fontColor':'white',
+                                          'showBackground':'true', 'alignment':'bottomCenter'})
+        
+        if 'atomLabel' in wgData:
+            
+            mol = molViewState.moldict[list(sMolNames)[0]]
+            confId=wgData['confId'].value
+            sconf = mol.GetConformer(confId)
+            xyz = sconf.GetPositions()
+            
+            if wgData['atomLabel'].value:
+                for i in list(range(sconf.GetNumAtoms())):
+                    atomLabel = '{}{}'.format(mol.GetAtomWithIdx(i).GetSymbol(), i)
+                    view.addLabel(atomLabel, {'inFront' : 'false', 
+                                              'fontSize' : '12',
+                                              'fontColor':'gray',
+                                              'showBackground':'false',
+                                              'position' : {'x':xyz[i][0], 'y':xyz[i][1], 'z':xyz[i][2]}
+                                             })
+    
+    return view
+    
+def handleProteinStyle(molViewState, view, pModelNum):
+    """ Handles protein drawing style in update3D function """
+    uid = molViewState.uid
+    wgData = globals()['wgData'][uid]
+    
+    # 'pStyle_wg' and 'pStyle_helicesAsTubes_wg' rendered as pair
+    if 'pStyle_wg' in wgData:
+        pStyle=wgData['pStyle_wg'].value
+    else:
+        pStyle = REPRESENTATION_DEFAULT['protein']
+        
+    
+    if pStyle == 'surface':
+        view.addSurface('SES', {'model':pModelNum});
+    elif pStyle == 'line':
+        view.setStyle({'model':pModelNum},{'line':{}});
+    elif pStyle == 'cartoon' and 'pStyle_helicesAsTubes_wg' not in wgData:
+        view.setStyle({'model':pModelNum},{'cartoon':{'color': 'spectrum', 'arrows': 'true'}})
+    else:
+        if wgData['pStyle_helicesAsTubes_wg'].value:
+            view.setStyle({'model':pModelNum},{'cartoon':{'color': 'spectrum',
+                                                          'arrows': 'true', 'tubes' : 'true'}})
+        else:
+            view.setStyle({'model':pModelNum},{'cartoon':{'color': 'spectrum', 'arrows': 'true'}})
+    
+    return view
+    
+    
+def update3D(model_id):
+    """ This function invoked whenever user interacts with widgets.
+    It runs first time through handle_button() when the start button clicked.
+    """
+    uid = globals()['wgMIds'][model_id]
```

Comment:
I have removed the model Id dependency. Kindly see the most recent commit. Thank you very much for advising on improvement.